examples/flutter_gallery/lib/welcome/home.dart

```diff
@@ -0,0 +1,169 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+
+import 'constants.dart';
+import 'steps/all.dart';
+import 'welcome_step_state.dart';
+
+const int _kSteps = 4;
+const int _kAutoProgressSeconds = 10;
+const int _kAutoProgressTransitionMilliseconds = 520;
+const Color footerColor = Color(0xffffffff);
+const TextStyle footerButtonTextStyle = TextStyle(
+  fontSize: 16.0,
+  fontWeight: FontWeight.bold,
+  color: kWelcomeBlue,
+);
+const double footerButtonHeight = 60.0;
+
+class Welcome extends StatefulWidget {
+  const Welcome({ Key key, this.onDismissed }) : super(key: key);
+  final VoidCallback onDismissed;
+
+  static const String routeName = '/welcome';
+
+  @override
+  _WelcomeState createState() => _WelcomeState();
+}
+
+class _WelcomeState extends State<Welcome> with TickerProviderStateMixin {
+  List<_WelcomeStep<StatefulWidget>> _steps;
+  TabController _tabController;
+  PageController _pageController;
+  Timer _autoProgressTimer;
+
+  final GlobalKey<FlutterWelcomeStepState> _initialStepKey = GlobalKey<FlutterWelcomeStepState>();
+  final GlobalKey<PlaygroundWelcomeStepState> _playgroundStepKey = GlobalKey<PlaygroundWelcomeStepState>();
+  final GlobalKey<DocumentationWelcomeStepState> _documentationStepKey = GlobalKey<DocumentationWelcomeStepState>();
+  final GlobalKey<ExploreWelcomeStepState> _exploreStepKey = GlobalKey<ExploreWelcomeStepState>();
+
+  @override
+  void initState() {
+    super.initState();
+    _steps = <_WelcomeStep<StatefulWidget>>[
+      _WelcomeStep<FlutterWelcomeStep>(_initialStepKey, FlutterWelcomeStep(key: _initialStepKey)),
```

Comment:
For this and customized stuff above, since you're every just going to call 3 things on the state, might as well create an interface instead of leaking knowledge of the specific type. Design with minimalized responsibilities.